commercetools-convenience/src/test/java/io/sphere/sdk/sequencegenerators/BigIntegerNumberGeneratorIntegrationTest.java

```diff
@@ -92,6 +101,44 @@ public void concurrentUsageTest() throws Exception {
         assertThat(numbersGenerated.containsAll(expectedNumbers)).isTrue();
     }
 
+    @Test
+    public void failWithNotConcurrentModificationException() {
+        final String container = randomKey();
+        final String key = randomKey();
+        final CustomObjectBigIntegerNumberGeneratorConfig config =
+                CustomObjectBigIntegerNumberGeneratorConfigBuilder.of(getSphereClient(getBadRequestHttpClient()), key)
+                        .container(container)
+                        .build();
+        final BigIntegerNumberGenerator generator = CustomObjectBigIntegerNumberGenerator.of(config);
+
+        Throwable thrown = catchThrowable(() -> { generator.getNextNumber().toCompletableFuture().join(); });
```

Comment:
this should also work: `catchThrowable(() -> generator.getNextNumber().toCompletableFuture().join());`